python/sglang/srt/mem_cache/hicache_storage.py

```diff
@@ -20,133 +24,390 @@ def get_hash_str(token_ids: List[int], prior_hash: Optional[str] = None) -> str:
+			)
+		]
+
+	def set(self,
+			key, 
+		 	value: Optional[Any] = None, 
+		 	target_location: Optional[Any] = None,
+			target_sizes: Optional[Any] = None) -> bool:
+		tensor_path = f"{self.file_path}/{key}.bin"
+		if self.exists(key):
+			logger.warning(f"Key {key} already exists. Skipped.")
+			return True
+		try:
+			torch.save(value, tensor_path)
+			return True
+		except Exception as e:
+			logger.error(f"Failed to save tensor {key}: {e}")
+			return False
+
+	def batch_set(self,
+			      keys,
+				  values: Optional[Any] = None,
+				  target_locations: Optional[Any] = None,
+				  target_sizes: Optional[Any] = None) -> bool:
+		for key, value in zip(keys, values):
+			if not self.set(key, value):
+				return False
+		return True
+
+	def delete(self, key: str) -> None:
+		tensor_path = f"{self.file_path}/{key}.bin"
+		try:
+			os.remove(tensor_path)
+		except FileNotFoundError:
+			logger.warning(f"Key {key} does not exist. Cannot delete.")
+			return
+
+	def exists(self, key) -> bool | dict:
+		tensor_path = f"{self.file_path}/{key}.bin"
+		return os.path.exists(tensor_path)
+
+	def clear(self) -> None:
+		try:
+			for filename in os.listdir(self.file_path):
+				file_path = os.path.join(self.file_path, filename)
+				if os.path.isfile(file_path):
+					os.remove(file_path)
+			logger.info("Cleared all entries in HiCacheFile storage.")
+		except Exception as e:
+			logger.error(f"Failed to clear HiCacheFile storage: {e}")
+
+@dataclass
+class MooncakeStoreConfig:
+	local_hostname: str
+	metadata_server: str
+	global_segment_size: int
+	local_buffer_size: int
+	protocol: str
+	device_name: str
+	master_server_address: str
+
+	@staticmethod
+	def from_file(file_path: str) -> "MooncakeStoreConfig":
+		"""Load the config from a JSON file."""
+		with open(file_path) as fin:
+			config = json.load(fin)
+		return MooncakeStoreConfig(
+			local_hostname=config.get("local_hostname"),
+			metadata_server=config.get("metadata_server"),
+			global_segment_size=config.get(
+				"global_segment_size"
+			),
+			local_buffer_size=config.get(
+				"local_buffer_size"
+			),
+			protocol=config.get("protocol", "tcp"),
+			device_name=config.get("device_name", ""),
+			master_server_address=config.get("master_server_address"),
+		)
+
+	@staticmethod
+	def load_from_env() -> "MooncakeStoreConfig":
+		"""Load config from a file specified in the environment variable."""
+		config_file_path = os.getenv("MOONCAKE_CONFIG_PATH")
+		if config_file_path is None:
+			raise ValueError(
+				"The environment variable 'MOONCAKE_CONFIG_PATH' is not set."
+			)
+		return MooncakeStoreConfig.from_file(config_file_path)
+
+
+class MooncakeStore(HiCacheStorage):
+	def __init__(self):
+		try:
+			from mooncake.store import MooncakeDistributedStore
+		except ImportError as e:
+			raise ImportError(
+				"Please install mooncake by following the instructions at "
+				"https://github.com/kvcache-ai/Mooncake/blob/main/doc/en/build.md "  # noqa: E501
```

Comment:
Let us use this link: https://kvcache-ai.github.io/Mooncake/getting_started/build.html